rollout/ephemeralmetadata.go

```diff
@@ -27,21 +27,19 @@ func (c *rolloutContext) reconcileEphemeralMetadata() error {
 
 	if fullyRolledOut {
 		// We are in a steady-state (fully rolled out). newRS is the stableRS. there is no longer a canary
-		err := c.syncEphemeralMetadata(ctx, c.newRS, stableMetadata)
-		if err != nil {
-			return err
-		}
-	} else {
-		// we are in a upgrading state. newRS is a canary
-		err := c.syncEphemeralMetadata(ctx, c.newRS, newMetadata)
-		if err != nil {
-			return err
-		}
-		// sync stable metadata to the stable rs
-		err = c.syncEphemeralMetadata(ctx, c.stableRS, stableMetadata)
-		if err != nil {
-			return err
-		}
+		return c.syncEphemeralMetadata(ctx, c.newRS, stableMetadata)
+	}
```

Comment:
you have changed the behaviour here with this early return. the for loop never happens down below.

Reply:
Thanks so much for your help, that was the issue I resolved it.